skypeweb/skypeweb_login.c

```diff
@@ -112,15 +112,17 @@ skypeweb_login_got_pie(PurpleUtilFetchUrlData *url_data, gpointer user_data, con
 	g_string_append_printf(postdata, "js_time=%" G_GINT64_FORMAT "&", skypeweb_get_js_time());
 	g_string_append(postdata, "client_id=578134&");
 	g_string_append(postdata, "redirect_uri=https://web.skype.com/");
+
+	if (postdata->len > INT_MAX) abort();

```

Comment:
Sure, but this is sort of an internal error if you can't fit the string length into INT_MAX and we need to fence against that. On second thought, how about just waiving error checking and clamping the output length instead? Something like this, with formatting adjusted:
```
int tmp = postdata -> len;
if (postdata -> len > INT_MAX) { tmp = INT_MAX; }
```

and then replace postdata->len in the format-string variable argument list by **tmp**.
Worst case this truncates output to 2 GB (32-bit machines, anything UNIX) which should be on the safe side (and even for 16-bit, 32768 bytes truncation would look pretty safe to me in login context for the next decades, I expect to see stronger, rather than longer, crypto, in those contexts).